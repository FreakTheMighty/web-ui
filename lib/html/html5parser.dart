// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.
// Generated by scripts/tree_gen.py.

/**
 * An HTML5 parser written in Dart.
 *
 * Intended to parse all html documents using the [html5 parsing algorithm][p],
 * although the implementation is not yet complete.
 *
 * [p]: http://dev.w3.org/html5/spec/parsing.html#parsing
 */
#library('html5parser');

#source('htmltree.dart');
#source('parser.dart');
#source('tree.dart');
#source('token.dart');
#source('tokenkind.dart');
#source('tokenizer_base.dart');
#source('tokenizer.dart');
