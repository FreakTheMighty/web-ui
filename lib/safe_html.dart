// Copyright (c) 2011, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

/** **Deprecated**: import 'web_ui/safe_html.dart' instead. */
@deprecated library safe_html;

import 'package:meta/meta.dart';
export 'package:web_ui/safe_html.dart';
