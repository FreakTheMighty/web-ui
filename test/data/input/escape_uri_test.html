<!DOCTYPE html>
<!--
Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
for details. All rights reserved. Use of this source code is governed by a
BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script type="application/javascript" src="testing.js"></script>
</head>
<body>
  <!-- Tests that you can inject any URL using SafeUri. -->
  Expect 1 console message: "m2", there should be no message for m1, m3, m4
  <a href="{{m1}}"></a>
  <a href="{{m2}}"></a>
  <a href="javascript:{{m3}}"></a>
  <a href="javascript:{{m4}}"></a>
  <script type="application/dart">
  import 'dart:html';
  import 'package:web_components/safe_html.dart';

  var m1 = 'javascript:window.console.log("m1")';
  var m2 = new SafeUri.unsafe('javascript:window.console.log("m2")');
  var m3 = 'window.console.log("m3")';
  var m4 = new SafeUri.unsafe('window.console.log("m4")');
  var html = new SafeHtml.unsafe('<span>hello</span>');
  main() {
    window.setTimeout(() {
      document.queryAll('a').forEach((tag) {
        tag.on.click.dispatch(
          new MouseEvent('click', window, 1, 0, 0, 0, 0, 0));
      });

      window.setTimeout(() => window.postMessage('done', '*'), 0);
    }, 0);
  }
  </script>
</body>
</html>
