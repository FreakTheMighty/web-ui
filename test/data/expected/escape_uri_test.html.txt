Blocked access to external URL https://dart.googlecode.com/svn/branches/bleeding_edge/dart/client/dart.js
Blocked access to external URL https://dart.googlecode.com/svn/branches/bleeding_edge/dart/client/dart.js
CONSOLE MESSAGE: line 1: m2
Content-Type: text/plain
layer at (0,0) size 800x600
  RenderView at (0,0) size 800x600
layer at (0,0) size 800x36
  RenderBlock {HTML} at (0,0) size 800x36
    RenderBody {BODY} at (8,8) size 784x20
      RenderText {#text} at (0,0) size 467x19
        text run at (0,0) width 467: "Expect 1 console message: \"m2\", there should be no message for m1, m3, m4"
      RenderInline {A} at (0,0) size 0x0 [color=#0000EE]
      RenderText {#text} at (0,0) size 0x0
      RenderInline {A} at (0,0) size 0x0 [color=#0000EE]
      RenderText {#text} at (0,0) size 0x0
      RenderInline {A} at (0,0) size 0x0 [color=#0000EE]
      RenderText {#text} at (0,0) size 0x0
      RenderInline {A} at (0,0) size 0x0 [color=#0000EE]
      RenderText {#text} at (0,0) size 0x0
      RenderText {#text} at (0,0) size 0x0
#EOF
